src/main/java/core/basesyntax/lib/Injector.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax.lib;
+
+import core.basesyntax.factory.Factory;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+
+public class Injector {
+
+    public static Object getInstance(Class clazz) throws IllegalAccessException,
+            InvocationTargetException, InstantiationException, NoSuchMethodException {
+
+        Constructor constructor = clazz.getConstructor();
+        Object instance = constructor.newInstance();
+
+        Field[] declaredFields = clazz.getDeclaredFields();
+
+        for (Field field : declaredFields) {
+            if (field.getAnnotation(Inject.class) != null
+                    && field.getName().equalsIgnoreCase("betDao")
```

Comment:
```suggestion
                    && field.getType().equals(BetDao.class)
```
краще тут не прив'язуватись до неймінгу поля а до його типу

Reply:
Fixed that, thanks :)